aws-cpp-sdk-s3-crt/source/S3CrtClient.cpp

```diff
@@ -201,6 +201,32 @@ void S3CrtClient::init(const S3Crt::ClientConfiguration& config, const std::shar
   aws_s3_client_config s3CrtConfig;
   AWS_ZERO_STRUCT(s3CrtConfig);
   s3CrtConfig.region = Aws::Crt::ByteCursorFromCString(config.region.c_str());
+  s3CrtConfig.connect_timeout_ms = config.connectTimeoutMs;
+
+  aws_s3_tcp_keep_alive_options *tcp_keep_alive_options = nullptr;
+  if (config.enableTcpKeepAlive)
+  {
+    const uint16_t keep_intvl = std::max<uint16_t>(15, config.tcpKeepAliveIntervalMs/1000);
+
+    tcp_keep_alive_options = static_cast<aws_s3_tcp_keep_alive_options *>(aws_mem_calloc(Aws::get_aws_allocator(),
```

Comment:
Do you need to allocate it when at the end of scope deallocate it? What about keep it on the stack?

This is the very same code
```
   aws_s3_tcp_keep_alive_options tcp_keep_alive_options;
   if (config.enableTcpKeepAlive)
   {
     const uint16_t keep_intvl = std::max<uint16_t>(15, config.tcpKeepAliveIntervalMs/1000);
     tcp_keep_alive_options.keep_alive_interval_sec = keep_intvl;
     tcp_keep_alive_options.keep_alive_timeout_sec = keep_intvl;

    s3CrtConfig.tcp_keep_alive_options = &tcp_keep_alive_options;
   }
```

Reply:
I used this in an earlier variant before submitting the PR. This part of the code was flagged by the address sanitizer, which flagged using the address of a local variable as an error. Hence switched to allocating it on the heap instead.

Reply:
Just re-ran the tests. It was a `stack-use-after-scope` error, and it was my fault, since I had declared the variable inside the `if`-block, like this:
```c++
   aws_s3_tcp_keep_alive_options tcp_keep_alive_options;
   if (config.enableTcpKeepAlive)
   {
     const uint16_t keep_intvl = std::max<uint16_t>(15, config.tcpKeepAliveIntervalMs/1000);
     tcp_keep_alive_options.keep_alive_interval_sec = keep_intvl;
     tcp_keep_alive_options.keep_alive_timeout_sec = keep_intvl;

    s3CrtConfig.tcp_keep_alive_options = &tcp_keep_alive_options;
   }
```
Will update the PR to use stack-allocated variables. Thank you for pointing this out.